src/Service/CSVReader.php

```diff
@@ -0,0 +1,103 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Service;
+
+use SplFileObject;
+use Symfony\Component\Serializer\Encoder\CsvEncoder;
+use Symfony\Component\Serializer\Serializer;
+
+class CSVReader implements ReaderInterface
+{
+    private int $batchSize; // Batch size in bytes
+
+    private array $report;
+
+    private SplFileObject $file;
+
+    private Serializer $serializer;
+
+    private int $batches;
+
+    private int $linesRead;
+
+    public function __construct()
+    {
+        $this->report = [];
+        $this->serializer = new Serializer([], [new CsvEncoder()]);
+        $this->batches = 0;
+        $this->linesRead = 0;
+    }
+
+    public function open(string $filename, int $batchSize, bool $noHeaders = false): ?string //if the file won't contain headers. $noHeaders = true
+    {
+        $this->batchSize = $batchSize;
+
+        if (!file_exists($filename)) {
```

Comment:
Big and complicated method
```suggestion
try {
    $this->validateFileExist($filename);
    $this->validateFileIsReadable($filename);
    $this->validateBatchSize($batchSize)
} catch(ReaderInputParameterException $e) {
    return $e->getMessage()
}
```
Moving it into separate Validator class even better

Reply:
Implemented.